src/peft/tuners/lora.py

```diff
@@ -1252,28 +1245,34 @@ def __init__(
         self.active_adapter = adapter_name
 
     def forward(self, x: torch.Tensor):
+        # note: logic differs from default Linear because merging is not supported
         result = self.quant_linear_module(x)
-        if self.disable_adapters or self.active_adapter not in self.lora_A.keys():
+
+        if (
+            self.disable_adapters
+            or (self.active_adapter not in self.lora_A.keys())
+            or (self.r[self.active_adapter] == 0)
+        ):
             return result
-        elif self.r[self.active_adapter] > 0:
-            result = result.clone()
-            if not torch.is_autocast_enabled():
-                expected_dtype = result.dtype
-                x = x.to(self.lora_A[self.active_adapter].weight.dtype)
-                output = (
-                    self.lora_B[self.active_adapter](
-                        self.lora_A[self.active_adapter](self.lora_dropout[self.active_adapter](x))
-                    ).to(expected_dtype)
-                    * self.scaling[self.active_adapter]
-                )
-            else:
-                output = (
-                    self.lora_B[self.active_adapter](
-                        self.lora_A[self.active_adapter](self.lora_dropout[self.active_adapter](x))
-                    )
-                    * self.scaling[self.active_adapter]
-                )
-            result += output
+
+        # FIXME: is it correct that we clone here but not for int8?
+        result = result.clone()
```

Comment:
Same comment as above, I don't think we should clone here, we do have GPTQ tests, maybe we can run the test and check if it works with and without `clone`?